src/Umbraco.Web/Models/Mapping/ContentMapperProfile.cs

```diff
@@ -88,12 +91,45 @@ internal class ContentMapperProfile : Profile
         }
     }
 
+    internal class CultureUpdateDateResolver : IValueResolver<IContent, ContentItemBasic<ContentPropertyBasic>, DateTime>
+    {
+        public DateTime Resolve(IContent source, ContentItemBasic<ContentPropertyBasic> destination, DateTime destMember, ResolutionContext context)
+        {
+            var culture = context.GetCulture();
+
+            //a culture needs to be in the context for a variant content item
+            if (culture == null)
+                return source.UpdateDate;
+
+            var pubDate = source.GetPublishDate(culture);
+            return pubDate.HasValue ? pubDate.Value : source.UpdateDate;
+        }
+    }
+
+    internal class CulturePublishedResolver : IValueResolver<IContent, ContentItemBasic<ContentPropertyBasic>, bool>
+    {
+        public bool Resolve(IContent source, ContentItemBasic<ContentPropertyBasic> destination, bool destMember, ResolutionContext context)
+        {
+            var culture = context.GetCulture();
+
+            //a culture needs to be in the context for a variant content item
+            if (culture == null)
+                return source.Published;
+
+            return source.IsCulturePublished(culture);
+        }
+    }
+
     internal class CultureNameResolver : IValueResolver<IContent, ContentItemBasic<ContentPropertyBasic>, string>
     {
         public string Resolve(IContent source, ContentItemBasic<ContentPropertyBasic> destination, string destMember, ResolutionContext context)
         {
             var culture = context.GetCulture();
 
+            //a culture needs to be in the context for a variant content item
+            if (culture == null)
+                return source.Name;
+            
             if (source.CultureNames.TryGetValue(culture, out var name) && !string.IsNullOrWhiteSpace(name))
```

Comment:
see other comment about .VariesByCulture(), same principal here, if it doesn't vary by culture then you need to just return `source.Name`